libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookAdapter.cs

```diff
@@ -133,7 +135,24 @@ public async Task ContinueConversationAsync(ConversationReference reference, Bot
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
         public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot bot, CancellationToken cancellationToken)
         {
-            if (!_facebookClient.VerifySignature(request))
+            if (request.Query["hub.mode"] == "subscribe")
+            {
+                await _facebookClient.VerifyWebhookAsync(request, response, cancellationToken).ConfigureAwait(false);
+                return;
+            }
+
+            await FacebookHelper.WriteAsync(response, HttpStatusCode.OK, string.Empty, Encoding.UTF8, cancellationToken).ConfigureAwait(false);
+
+            string body;
+
+            using (var sr = new StreamReader(request.Body))
+            {
+                body = sr.ReadToEnd();
+            }
+
+            var facebookBody = JsonConvert.SerializeObject(body);
```

Comment:
facebookBody doesn't seem to be used anywhere else, but I understand this is a WIP

Reply:
You are right. I'll remove it.

Reply:
Done. Thanks!